src/arrays.jl

```diff
@@ -0,0 +1,90 @@
+struct ArrayPlan{T, FF<:FTPlan{<:T}, Szs<:Tuple, Dims<:Tuple{<:Int}} <: Plan{T}
+    F::FF
+    szs::Szs
+    dims::Dims
+end
+size(P::ArrayPlan) = P.szs
+size(P::ArrayPlan, k::Int) = P.szs[k]
+size(P::ArrayPlan, k...) = P.szs[[k...]]
+
+function ArrayPlan(F::FTPlan{<:T}, c::AbstractArray{T}, dims::Tuple{<:Int}=(1,)) where T
+    szs = size(c)
+    @assert F.n == szs[dims[1]]
+    ArrayPlan(F, size(c), dims)
+end
+
+function *(P::ArrayPlan, f::AbstractArray)
+    F, dims, szs = P.F, P.dims, P.szs
+    @assert length(dims) == 1
+    @assert szs == size(f)
+    d = first(dims)
+
+    perm = (d, ntuple(i-> i + (i >= d), ndims(f) -1)...)
+    fp = permutedims(f, perm)
+
+    fr = reshape(fp, size(fp,1), :)
+
+    permutedims(reshape(F*fr, size(fp)...), invperm(perm))
+end
+
+function \(P::ArrayPlan, f::AbstractArray)
+    F, dims, szs = P.F, P.dims, P.szs
+    @assert length(dims) == 1
+    @assert szs == size(f)
+    d = first(dims)
+
+    perm = (d, ntuple(i-> i + (i >= d), ndims(f) -1)...)
+    fp = permutedims(f, perm)
+
+    fr = reshape(fp, size(fp,1), :)
+
+    permutedims(reshape(F\fr, size(fp)...), invperm(perm))
+end
+
+struct NDimsPlan{T, FF<:ArrayPlan{<:T}, Szs<:Tuple, Dims<:Tuple} <: Plan{T}
+    F::FF
+    szs::Szs
+    dims::Dims
+    function NDimsPlan(F, szs, dims)
+        if length(Set(szs[[dims...]])) > 1
+            error("Different size in dims axes not yet implemented in N-dimensional transform.")
+        end
+        new{eltype(F), typeof(F), typeof(szs), typeof(dims)}(F, szs, dims)
+    end
+end
+
+size(P::NDimsPlan) = P.szs
+size(P::NDimsPlan, k::Int) = P.szs[k]
+size(P::NDimsPlan, k...) = P.szs[[k...]]

```

Comment:
Why the array construction in the indexing? Shouldn't this just be `P.szs[k...]`?

Reply:
Perhaps I've misunderstood this. Are you trying to obtain multiple indices in one go?

Reply:
That's the intention! Although I don't need that functionality in this PR.

Reply:
Perhaps best to remove it then, as it's not the conventional usage of `size`. One may broadcast `size` over indices to obtain the sizes along multiple dimensions at once.

Reply:
Note sizes of plans and arrays behave differently 

compare w FFT

Reply:
judging by this line here https://github.com/JuliaMath/FFTW.jl/blob/f888022d7a1ff78491abf8f33f1055cc52a68f0a/src/fft.jl#L254 I think I should only really keep `size(P::NDimsPlan) = P.szs`

Reply:
...or maybe I should keep `size(P::NDimsPlan, k::Int) = P.szs[k]`?